### PR TITLE
8 addcommnets

### PR DESCRIPTION
**Pull request by Nicolakirk**
add comments added

### CODE REVIEW

File: src/components/CommentsForm.jsx [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+   
+    setUserName("tickle122");
+
+    const inputComment = {
+    
+    username: userName,
+         body: body,
+         votes:0,
+                    };
+                console.log(inputComment);
```
Comment (itskatherine) [inline src/components/CommentsForm.jsx]: console log begone

File: src/components/CommentsForm.jsx [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+                  setCommentsList((currentComments)=> {
+                    return ([inputComment, ...currentComments])})
+                    setMessage("We have updated your comment");
+                     setErr(null)
+ postComments(articleId, inputComment).catch((err) => {
+    if(body === ""){
+        setMessage("Please enter a comment")}
+        else{
+            setCommentsList((currentComments) => { 
+       (currentComments.shift()); })
```
Comment (itskatherine) [inline src/components/CommentsForm.jsx]: We should never mutate the state in these functions, you should make a copy of the currentComments and alter that instead 👍

File: src/components/CommentsForm.jsx [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+const [err, setErr] =useState(null);
+
+const[message, setMessage]= useState("");
+
+
+
+const handleSubmit =(event)=>{
+    event.preventDefault();
+   
+    setUserName("tickle122");
```
Comment (itskatherine) [inline src/components/CommentsForm.jsx]: There's no point at which you change this state so you don't need this line.
Strictly as this isn't changed anywhere it doesn't need to be in a state 🤔

File: src/components/CommentList.jsx [hunk 4]
```diff
@@ -20,25 +25,36 @@ const CommentList=()=> {
...
+      }, [articleId]);
+    
       if(isLoading ){
         return (
           <p>Is loading ....</p>
         )
       }
+  
+console.log(commentsList);
+console.log(commentsList[1])
```
Comment (itskatherine) [inline src/components/CommentList.jsx]: console logs begone

File: src/components/CommentsForm.jsx [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+
+    const inputComment = {
+    
+    username: userName,
+         body: body,
+         votes:0,
+                    };
+                console.log(inputComment);
+                  setCommentsList((currentComments)=> {
+                    return ([inputComment, ...currentComments])})
```
Comment (itskatherine) [inline src/components/CommentsForm.jsx:33]: Nice use of optimistic rendering

**Review by itskatherine — CHANGES_REQUESTED** [review]
Comment adder looking good Nicola, I like your use of optimistic rendering and your CSS 🎨 
The layout of your code makes it quite difficult to read - I'd consider using a VScode extension like Prettier to format your code. 💡 
Also atm I'm able to spam multiple empty comments (they dont persist on the backend but remain on the screen in state), can you prevent the user doing this? 
Can you also add functionality so that a user can't post a comment while the previous comment is in the process of being posted by the api? Once you've added these please resubmit 📝

File: src/components/CommentsForm.jsx [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+            <h3 className="grid_item1" >Post a New Comment Here</h3>
+          
+              <input className="grid_item2"
+              type="text"
+              value={body}
+              maxLength="1000"
+              required
+              disabled={submitting}
+              onChange={(event)=> setBody(event.target.value)}
+              />
```
Comment (popatre) [inline src/components/CommentsForm.jsx]: Small thing, but could you change this to a textarea to enable multi line comments - just incase someone wants to post a bit more of a comment

File: src/components/CommentsForm.jsx [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+       <div className="commentsform" >  <form
+            
+            onSubmit={handleSubmit}> 
+            <h3 className="grid_item1" >Post a New Comment Here</h3>
+          
+              <input className="grid_item2"
+              type="text"
+              value={body}
+              maxLength="1000"
+              required
```
Comment (popatre) [inline src/components/CommentsForm.jsx]: Nice controlled component

**Review by popatre — APPROVED** [review]
Lovely stuff - 
Small thing, but could you change this to a textarea to enable multi line comments - just incase someone wants to post a bit more of a comment.
Good to merge after that though

File: src/components/CommentsForm.jsx (hunk 7 above)
Comment (Nicolakirk) [reply src/components/CommentsForm.jsx]: Thanks

**Pull request merged**